src/CpmsCommon/Queue/Adapter/Synchronous/SynchronousQueueAdapterFactory.php

```diff
@@ -2,32 +2,29 @@
 
 namespace CpmsCommon\Queue\Adapter\Synchronous;
 
-use Interop\Container\ContainerInterface;
 use Interop\Container\Exception\ContainerException;
 use Laminas\ServiceManager\Exception\ServiceNotCreatedException;
 use Laminas\ServiceManager\Exception\ServiceNotFoundException;
-use Laminas\ServiceManager\Factory\FactoryInterface;
+use Laminas\ServiceManager\ServiceLocatorInterface;
 
 /**
  * Class SynchronousQueueAdapterFactory
  *
  * @package Queue\Adapter\Immediate
  */
-class SynchronousQueueAdapterFactory implements FactoryInterface
+class SynchronousQueueAdapterFactory
```

Comment:
Why the change here? I think every Factory should follow this pattern

Reply:
Because the `setServiceContainer` call on the body of the factory needs a ServiceLocatorInterface (vendor/laminas/laminas-servicemanager/src/ServiceLocatorInterface.php), which is a child of the `ContainerInterface`. So the signatures don't match.